dask_geopandas/io/arrow.py

```diff
@@ -0,0 +1,374 @@
+
+        _kwargs = kwargs.get("arrow_to_pandas", {})
+        _kwargs.update({"use_threads": False, "ignore_metadata": False})
+
+        # TODO support additional keywords
+        try:
+            return _arrow_to_geopandas(arrow_table)
+        except ValueError as err:
+            # when no geometry column is selected, the above will error.
+            # We want to fallback to reading it as a plain dask object, because
+            # the column selection can be an automatic pushdown (eg `ddf['col']`)
+            # TODO more robust detection of when to fall back?
+            if "No geometry columns are included" in str(err):
+                return super()._arrow_table_to_pandas(
+                    arrow_table, categories=categories, **kwargs
+                )
+            else:
+                raise
+
+    @classmethod
+    def _pandas_to_arrow_table(
+        cls, df: pd.DataFrame, preserve_index=False, schema=None
+    ) -> "pyarrow.Table":
+        from geopandas.io.arrow import _geopandas_to_arrow
+
+        # TODO add support for schema
+        if schema is not None:
+            raise NotImplementedError("Passing 'schema' is not yet supported")
+
+        table = _geopandas_to_arrow(df, index=preserve_index)
+        return table
+
+
+class FeatherDatasetEngine(GeoDatasetEngine, ArrowDatasetEngine):
+    file_format = "feather"
+
+
+class FeatherFunctionWrapper:
+    """
+    Feather Function-Wrapper Class
+    Reads Feather data from disk to produce a partition.
+    """
+
+    def __init__(self, engine, fs, columns, filter, schema, index):
+        self.engine = engine
+        self.fs = fs
+        self.columns = columns
+        self.filter = filter
+        self.schema = schema
+        self.index = index
+
+    def project_columns(self, columns):
+        """Return a new FeatherFunctionWrapper object with
+        a sub-column projection.
+        """
+        if columns == self.columns:
+            return self
+        func = copy.deepcopy(self)
+        func.columns = columns
+        return func
+
+    def __call__(self, parts):
+        _df = self.engine.read_partition(
+            self.fs, parts, self.schema, self.columns, self.filter
+        )
+        if self.index:
+            _df.set_index(self.index, inplace=True)
+        return _df
+
+
+def read_feather(
+    path,
+    columns=None,
+    filters=None,
+    index=None,
+    storage_options=None,
+):
+    """Read a Feather dataset into a Dask-GeoPandas DataFrame.
+
+    Parameters
+    ----------
+    path: str or list(str)
+        Source directory for data, or path(s) to individual Feather files.
+        Paths can be a full URL with protocol specifier, and may include
+        glob character if a single string.
+    columns: None or list(str)
+        Columns to load. If None, loads all.
+    index: str
+        Column name to set as index.
+    storage_options: None or dict
+        Further parameters to pass to the bytes backend.
```

Comment:
It might be a good idea to add in a link or suggestion about where to go for more information about this parameter.